multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ModuleDependencyChecker.java

```diff
@@ -1,74 +1,76 @@
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
+import org.cloudfoundry.multiapps.controller.core.Messages;
+import org.cloudfoundry.multiapps.controller.core.helpers.ModuleToDeployHelper;
+import org.cloudfoundry.multiapps.controller.core.util.UserMessageLogger;
 import org.cloudfoundry.multiapps.mta.model.Module;
 
 import com.sap.cloudfoundry.client.facade.CloudControllerClient;
-import com.sap.cloudfoundry.client.facade.domain.CloudApplication;
 
 public class ModuleDependencyChecker {
 
+    private final Map<String, Module> allModulesInDescriptorWithNames;
     private final Set<String> modulesForDeployment;
     private final Set<String> modulesNotForDeployment;
     private final Set<String> modulesAlreadyDeployed;
     private final CloudControllerClient client;
+    private final ModuleToDeployHelper moduleToDeployHelper;
+    private final UserMessageLogger userMessageLogger;
 
-    public ModuleDependencyChecker(CloudControllerClient client, List<Module> allModulesInDescriptor, List<Module> allModulesToDeploy,
-                                   List<Module> completedModules) {
+    public ModuleDependencyChecker(CloudControllerClient client, UserMessageLogger userMessageLogger,
+                                   ModuleToDeployHelper moduleToDeployHelper, List<Module> allModulesInDescriptor,
+                                   List<Module> allModulesToDeploy, List<Module> completedModules) {
         this.client = client;
-
-        modulesForDeployment = computeModulesForDeployment(allModulesToDeploy);
-        modulesAlreadyDeployed = computeAlreadyDeployedModules(completedModules);
-        modulesNotForDeployment = computeModuleNotForDeployment(allModulesInDescriptor);
+        this.userMessageLogger = userMessageLogger;
+        this.moduleToDeployHelper = moduleToDeployHelper;
+        allModulesInDescriptorWithNames = computeAllModulesMap(allModulesInDescriptor);
+        modulesForDeployment = computeModules(allModulesToDeploy);
+        modulesAlreadyDeployed = computeModules(completedModules);
+        modulesNotForDeployment = computeModules(allModulesInDescriptor, module -> !modulesForDeployment.contains(module));
     }
 
-    private Set<String> computeModulesForDeployment(List<Module> allModulesToDeploy) {
-        return allModulesToDeploy.stream()
-                                 .map(Module::getName)
-                                 .collect(Collectors.toSet());
+    private Map<String, Module> computeAllModulesMap(List<Module> allModules) {
+        return allModules.stream()
+                         .collect(Collectors.toMap(Module::getName, Function.identity()));
     }
 
-    private Set<String> computeAlreadyDeployedModules(List<Module> completedModules) {
-        return completedModules.stream()
-                               .map(Module::getName)
-                               .collect(Collectors.toSet());
+    private Set<String> computeModules(List<Module> modules) {
```

Comment:
Why the name of method was changed. I am little bit confused because there are two methods with the same name. From the names I couldn't understand what should be do the first one and the second one.

Reply:
Before this change `computeModulesForDeployment` and `computeAlreadyDeployedModules` were basically the same. `computeModuleNotForDeployment` was very similar, but contained additional filtering. I decided to combine the three methods into one generic.